docs/templates/plugin.rst.j2

```diff
@@ -129,6 +129,13 @@ Parameters
                     {% elif value.default is sameas false %}
                         {% set _x = value.update({'default': 'no'}) %}
                     {% endif %}
+                    {% if value.default is string %}
```

Comment:
can you turn `value.default` into a `bool` right before previous if/elif block instead?

Reply:
ok, that's a better way.

Reply:
I go through the code again, and the previous block does "bool -> string" staff, so I think I should not do "string -> bool" before a "bool -> string". And there is not an easy way to convert "yes" or "yeS" to True in jinja2. Could I remain this unchanged?

Reply:
All vars should go through normalization process first to be eligible for quick and easy comparison. That's why the block above turns everything into strings.

But before that, there are some initial strings and we need to *verify* what logical values they correspond to, so we should first convert all those possible cases like `YeS`, `nO`, `true`, `FalSe`, which could originally come from boolean or string literals to some common type, which in this case is boolean. This layer is separate from jinja, and should happen before.

Now, in jinja need to render strings, so we convert that to strings just for jinja.

So the common flow is:
YAML -> var_type_for_coding (python) -> var_type_for_renderring (jinja)

The problem is there within yaml.
In yaml you can write:
```yaml
var: true
```
and it will return boolean `True` in python.
for 
```yaml
var: yes
```
it will return `"yes"` of `str` type.

So now we need to turn all of the possible spellings of truthful values into boolean, so that in python in would be only `True` or `False`.

Then, in jinja we need to represent that as `"yes"` or `"no"`. It's an ideal case and it works like this within ansible.

But docs generator (which is a separate piece of software) looks to be skipping this normalization step, effectively bypassing `Yes`/`yes` as a string, so that only if there were real YAML boolean literals `true`/`false` they are converted into `yes` or `no`. But strings get there unprocessed.

So by adding this lowercasing of supposingly boolean vars you work around design issue, "fixing" side effects, instead of fixing the root of the issue, the main cause of such behavior.